src/Bicep.Core/Configuration/ExperimentalFeaturesEnabledConfiguration.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Immutable;
+using System.Text.Json;
+
+namespace Bicep.Core.Configuration;
+
+public class ExperimentalFeaturesEnabledConfiguration : ConfigurationSection<ImmutableDictionary<string, bool>>
```

Comment:
Since we have a known set of features, why not represent this as a class with bool properties?